client-haskell/src/Icepeak/Client.hs

```diff
@@ -37,28 +48,49 @@ data Client = Client
   , clientAuth :: ByteString
   }
 
--- | Set a value at the leaf of a path.
---
+-- $updatebehavior
 -- Returns the status code of the HTTP response. Icepeak returns 202 if the
 -- update was accepted, and 503 if the high water mark was reached. Any other
 -- status code indicates a different error, or a misconfigured proxy.
 --
 -- Will rethrow any exceptions thrown by the I/O actions from
 -- "Network.HTTP.Client". Will not throw any other exceptions.
+
+-- | Set a value at the leaf of a path.
 setAtLeaf :: (MonadIO m, ToJSON a) => HTTP.Manager -> Client -> [Text] -> a -> m HTTP.Status
 setAtLeaf httpManager client path leaf =
   let request = setAtLeafRequest client path leaf
   in liftIO . fmap HTTP.responseStatus $ HTTP.httpNoBody request httpManager
 
+-- | Delete the value at the leaf of a path.
+deleteAtLeaf :: MonadIO m => HTTP.Manager -> Client -> [Text] -> m HTTP.Status
```

Comment:
Is there a concrete advantage of having this be an `MondadIO m` instead of just being `IO`?
Does servant roll its own IO monad or so?

Reply:
So that the caller does not need to use `liftIO`.